### DISCUSSION TITLE
Exercise 8: What good is a distributed snapshot when the system was never in the state represented by the distributed snapshot? Give an application of distributed snapshots.

### DISCUSSION THREAD

**Discussion by dag7dev** [sapienzastudentsnetwork/distributed-systems#33] [Exercises]
**Exercise 8:** What good is a distributed snapshot when the system was never in the state represented by the distributed snapshot? Give an application of distributed snapshots.

**Comment by riccardoscuto** [comment]
From the Drive 

Assuming that the channels are FIFO, when a process receives its first "take snapshot" message it records its state and broadcasts the message to every other process. The protocol terminates after the process has received its last "take snapshot" message. Since we assumed that the channels are FIFO, it cannot happen that a message sent after recording its state is received before the "take snapshot" message from the same process. Thus, the global state $S$ is consistent.

**Comment by susannacifani** [comment]
> From the Drive
> 
> Assuming that the channels are FIFO, when a process receives its first "take snapshot" message it records its state and broadcasts the message to every other process. The protocol terminates after the process has received its last "take snapshot" message. Since we assumed that the channels are FIFO, it cannot happen that a message sent after recording its state is received before the "take snapshot" message from the same process. Thus, the global state $S$ is consistent.

I'm not sure about this one, I think it's correct but it should be explained better.
What we can conclude is that "A distributed snapshot is a useful tool for capturing the consistent state of a distributed system at a given moment" or am I wrong?

**Comment by Tommaso-Sgroi** [comment]
> > From the Drive
> > Assuming that the channels are FIFO, when a process receives its first "take snapshot" message it records its state and broadcasts the message to every other process. The protocol terminates after the process has received its last "take snapshot" message. Since we assumed that the channels are FIFO, it cannot happen that a message sent after recording its state is received before the "take snapshot" message from the same process. Thus, the global state $S$ is consistent.
> 
> I'm not sure about this one, I think it's correct but it should be explained better. What we can conclude is that "A distributed snapshot is a useful tool for capturing the consistent state of a distributed system at a given moment" or am I wrong?

I asked for a better explanation of this question to the professor, he said that it's a more "philosophical" question (whatever it means). He also said that require a more technical answer (again... whatever it means).
